source/IntegrationTests/Behaviours/IncomingRequests/GivenAggregatedMeasureDataRequestWithDelegationTests.cs

```diff
@@ -0,0 +1,926 @@
+[SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1201:Elements should appear in the correct order", Justification = "Test class")]
+public class GivenAggregatedMeasureDataRequestWithDelegationTests : BehavioursTestBase
+{
+    public GivenAggregatedMeasureDataRequestWithDelegationTests(IntegrationTestFixture integrationTestFixture, ITestOutputHelper testOutputHelper)
+        : base(integrationTestFixture, testOutputHelper)
+    {
+    }
+
+    public static object[][] DocumentFormatsWithAllRoleCombinations() => DocumentFormatsWithRoleCombinations(false);
+
+    public static object[][] DocumentFormatsWithRoleCombinationsForNullGridArea() => DocumentFormatsWithRoleCombinations(true);
+
+    public static object[][] DocumentFormatsWithRoleCombinations(bool nullGridArea)
+    {
+        var roleCombinations = new List<(ActorRole DelegatedFrom, ActorRole DelegatedTo)>
+        {
+            // Energy supplier, metered data responsible and balance responsible can only delegate to delegated
+            (ActorRole.EnergySupplier, ActorRole.Delegated),
+            (ActorRole.BalanceResponsibleParty, ActorRole.Delegated),
+        };
+
+        // Grid operator and MDR cannot make request with null grid area
+        if (!nullGridArea)
+        {
+            roleCombinations.Add((ActorRole.MeteredDataResponsible, ActorRole.Delegated));
+            roleCombinations.Add((ActorRole.GridOperator, ActorRole.Delegated));
+
+            // Grid operator can delegate to both delegated and grid operator
+            roleCombinations.Add((ActorRole.GridOperator, ActorRole.GridOperator));
+        }
+
+        var requestDocumentFormats = DocumentFormats
+            .GetAllDocumentFormats(except: new[]
+            {
+                DocumentFormat.Ebix.Name, // ebIX is not supported for requests
+            })
+            .ToArray();
+
+        var peekDocumentFormats = DocumentFormats.GetAllDocumentFormats();
+
+        return roleCombinations.SelectMany(d => requestDocumentFormats
+                .SelectMany(
+                    incomingDocumentFormat => peekDocumentFormats
+                        .Select(
+                            peekDocumentFormat => new object[]
+                            {
+                                incomingDocumentFormat,
+                                peekDocumentFormat,
+                                d.DelegatedFrom,
+                                d.DelegatedTo,
+                            })))
+            .ToArray();
+    }
+
+    [Theory]
+    [MemberData(nameof(DocumentFormatsWithAllRoleCombinations), MemberType = typeof(GivenAggregatedMeasureDataRequestWithDelegationTests))]
+    public async Task AndGiven_DelegationInOneGridArea_When_DelegatedActorPeeksAllMessages_Then_ReceivesOneNotifyAggregatedMeasureDataWithCorrectContent(DocumentFormat incomingDocumentFormat, DocumentFormat peekDocumentFormat, ActorRole delegatedFromRole, ActorRole delegatedToRole)
```

Comment:
```suggestion
    public async Task AndGiven_DelegationInOneGridArea_When_DelegatedActorPeeksAllMessages_Then_ReceivesOneNotifyAggregatedMeasureDataWithCorrectContent(DocumentFormat incomingDocumentFormat, DocumentFormat peekDocumentFormat, ActorRole delegatedFromRole, ActorRole delegatedToRole)
```

I see two deligations, but a request in one grid area? I dont know if there is a better name for this scenario?